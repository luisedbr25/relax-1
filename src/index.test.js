import { createStore } from './index';

/** Actions */
const updateUser = (state, next) => ({
    user: {
        ...state.user,
        ...next,
    },
});

const updateName = name => ({ state }) => ({
    _action: 'updateName',
    ...updateUser(state, {
        name,
    }),
});

const fetchData = () => ({ dispatch }) => (
    Promise.resolve(dispatch(updateName('Relax Is Amazing!'))).then(result => ({
        _action: 'fetchData',
        ...result,
    }))
);

/** Test */
describe('Relax', () => {
    it('basic setup', () => {
        const store = createStore();
        expect(store).toMatchSnapshot();
    });

    it('initial state', () => {
        const initialState = {
            name: 'Relax',
        };
        const store = createStore({ initialState });
        expect(store.getState()).toMatchSnapshot(initialState);
        expect(store.getState().name).toEqual('Relax');
    });

    it('Dispatch Sync', () => {
        const store = createStore();
        store.dispatch(updateName('Relax Is Awesome!')).then(({ state }) => {
            expect(state.user.name).toEqual('Relax Is Awesome!');
        });
        expect(store.getState()).toMatchSnapshot();
        expect(store.getState().user.name).toEqual('Relax Is Awesome!');
    });

    it('Nest Dispatch Async', () => {
        const store = createStore();
        store.dispatch(fetchData()).then(({ state }) => {
            expect(state.user.name).toEqual('Relax Is Amazing!');
        });
        expect(store.getState()).toMatchSnapshot();
        expect(store.getState().user.name).toEqual('Relax Is Amazing!');
    });
});
