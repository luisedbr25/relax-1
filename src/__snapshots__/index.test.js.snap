// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Relax Dispatch Sync 1`] = `
Object {
  "user": Object {
    "name": "Relax Is Awesome!",
  },
}
`;

exports[`Relax Nest Dispatch Async 1`] = `
Object {
  "user": Object {
    "name": "Relax Is Amazing!",
  },
}
`;

exports[`Relax basic setup 1`] = `
Object {
  "callSubscribers": [Function],
  "dispatch": [Function],
  "getState": [Function],
  "setState": [Function],
  "subscribe": [Function],
  "subscribers": Array [],
  "unsubscribe": [Function],
}
`;

exports[`Relax initial state 1`] = `
Object {
  "name": "Relax",
}
`;
